graphql_compiler/cost_estimation/filter_selectivity_utils.py

```diff
@@ -44,6 +112,216 @@ def _are_filter_fields_uniquely_indexed(filter_fields, unique_indexes):
     return False
 
 
+def _convert_uuid_string_to_int(uuid_string):
+    """Return the integer representation of a UUID string."""
+    return UUID(uuid_string).int
+
+
+def _is_IntegerInterval_empty(interval):
+    """Return True if the given interval is empty."""
```

Comment:
This is a bit confusing. An `IntegerInterval` represents a non-empty range, so of course it's not empty. I see a couple of ways to resolve this:
1. Rephrase this function as a validation, that checks if an `IntegerInterval` is valid. I'm not a fan of this, because the name would become more obscure. Also you're not using it for that purpose.
2. Remove the restriction that an interval has to be non-empty, and feel free to pass around (5, 3) as a valid interval containing 0 integers. I'm also not a fan because you create a foot-gun. Someone might mistakenly use `upper_bound - lower_bound + 1` as a size formula.
3. Make a `make_integer_interval(lower_bound, upper_bound)` that returns `None` if the interval would be empty.

Reply:
Thanks for the list, it was very helpful. I've decided on 3).